pybaseball/lahman.py

```diff
@@ -33,109 +38,109 @@ def download_lahman():
 def parks():
 	# do this for every table in the lahman db so they can exist as separate functions
 	z = get_lahman_zip()
-	f = os.path.join(base_string, "Parks.csv")
+	f = get_base_string(_separator) + _separator + "Parks.csv"
```

Comment:
I have a _feeling_ like this should be doable more cleanly, can you show me your original bug? I might have an idea for this.

Reply:
I hadn't experienced it before because I had downloaded the lahman files, but I recently deleted them and now running the marcel unit tests fail on my Windows machine, all with errors similar to this one:

```
tests\pybaseball\analysis\projections\marcels\test_marcels_pitching.py:17:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pybaseball\analysis\projections\marcels\marcels_base.py:35: in __init__
    self.stats_df = stats_df if stats_df is not None else self._load_data()
pybaseball\analysis\projections\marcels\marcels_pitching.py:19: in _load_data
    return pitching()
pybaseball\lahman.py:155: in pitching
    data = pd.read_csv(f if z is None else z.open(f), header=0, sep=',', quotechar="'")
C:\Python38\lib\zipfile.py:1514: in open
    zinfo = self.getinfo(name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _  

self = <zipfile.ZipFile file=<_io.BytesIO object at 0x000001B95CE3F9F0> mode='r'>
name = 'baseballdatabank-master\\core\\Pitching.csv'

    def getinfo(self, name):
        """Return the instance of ZipInfo given 'name'."""
        info = self.NameToInfo.get(name)
        if info is None:
>           raise KeyError(
                'There is no item named %r in the archive' % name)
E           KeyError: "There is no item named 'baseballdatabank-master\\\\core\\\\Pitching.csv' in the archive"

C:\Python38\lib\zipfile.py:1441: KeyError
```

Reply:
Now that I think about it, I have a more clean solution in my branch where I've been working on caching. I can bring that one in if you'd like.

Reply:
actually could the lahman bug fix be moved to a separate PR? I know it's a minor point but I feel like it'd help in focusing discussion on the more substantive changes here

Reply:
Sure, gimme just a few.

Reply:
It's in #116

Reply:
Merged into this branch as well.